### ISSUE TITLE
`jsdoc/no-undefined-types` throws error for local class properties when documenting the class

### ISSUE THREAD

**Issue by PartyWumpus** [gajus/eslint-plugin-jsdoc#1762]
## Expected behavior

I should be able to reference a property on the current class I'm documenting (as in when writing a docstring _on_ the class, not _in_ the class) without using the class' full name. (as in `{@link property_a}` instead of `{@link MyClassName.property_a}`. (Typescript understands the annotation)

## Actual behavior

Raises an error `jsdoc/no-undefined-types: The type 'property' is undefined` on the line where the doc is.

## ESLint Config

```js
'jsdoc/no-undefined-types': 'error'
```

## ESLint sample

```typescript
/** xyz {@link property} */
class MyClass {
	public property: boolean
}
```

## Environment

- Node version: v24.19.0
- ESLint version v10.9.1
- `eslint-plugin-jsdoc` version: 64.3.7

**Comment by PartyWumpus** [comment]
follow on from https://github.com/gajus/eslint-plugin-jsdoc/issues/1760

**Comment by brettz9** [comment]
I've filed #1763 to fix. We try to stick to just one release per 24 hours, so will wait a little bit on this.
